src/cli/evaluations/summarize.py

```diff
@@ -0,0 +1,99 @@
+
+METRICS = ["mase", "smape", "nrmse", "nd", "ncrps"]
+
+DATASETS = ["m3_yearly", "m3_quarterly", "m3_monthly", "m3_other", "m4_quarterly", "m4_monthly", 
+    "m4_weekly", "m4_daily", "m4_hourly", "m4_yearly", "tourism_quarterly", "tourism_monthly", 
+    "dominick", "weather", "hospital", "covid_deaths", "electricity", "kdd_2018", "nn5", "rossmann", "solar", "taxi", "wiki"]
+
+@evaluations.command(
+    short_help="Visulize results as a table."
+)
+@click.option(
+    "--evaluations_path",
+    type=click.Path(exists=True),
+    default=DEFAULT_EVALUATIONS_PATH,
+    help="The directory where TSBench evaluations are stored.",
+)
+@click.option(
+    "--experiment",
+    type=click.Path(),
+    required=True,
+    help="The experiment name which you want to visualization.",
+)
+@click.option(
+    "--metric",
+    type=click.Path(),
+    default='ncrps',
+    help="The metric score shown in table.",
+)
+def summarize(evaluations_path: Path, experiment: str, metric:str):
+    results = []
+    source = Path(evaluations_path)
+    source = Path.joinpath(source, experiment)
+    if Path.joinpath(source, experiment + '.csv').exists():
+        print("load from csv file:", Path.joinpath(source, experiment + '.csv'))
+        res_df = pd.read_csv(Path.joinpath(source, experiment + '.csv'))
+        index_models = list(set(res_df.model.values.tolist()))
+    else:
+        models = os.listdir(source)
+        autogluon_models = set()
+        for model in models:
+            model_dir = Path.joinpath(source, model)
+            if Path.is_file(model_dir):
+                continue
+            datasets = os.listdir(model_dir)
+            for ds in datasets:
+                # TODO collect dataset we need, try collect all dataset but just print we need
+                if ds not in DATASETS:
+                    continue
+                ds_dir = Path.joinpath(model_dir, ds)
+                hyperparameters = os.listdir(ds_dir)
+                for hp in hyperparameters:
+                    hp_dir = Path.joinpath(ds_dir, hp)
+                    config = json.load(open(Path.joinpath(hp_dir, 'config.json'), 'r'))
+                    performance = json.load(open(Path.joinpath(hp_dir, 'performance.json'), 'r'))
+                    n = len(performance['performances'])
+                    res = {}
+                    if model == 'autogluon':
+                        autogluom_model = model + '-' + config['hyperparameters']['presets'] + '-' + str(config['hyperparameters']['run_time'])
+                        autogluon_models.add(autogluom_model)
+                        res['model'] = autogluom_model
+                    else:
+                        res['model'] = model
+                    res['dataset'] = ds
+                    res.update(performance['performances'][-1]['testing'])
+                    val_loss = performance['performances'][n-1]['evaluation']['val_loss'] if 'evaluation' in performance['performances'][n-1] else -1
```

Comment:
Could you add a comment on why this is needed?

Reply:
Also, here we assume we always take the last model. Actually when there are multiple epochs, we may want to take the iteration with the minimum val_loss.